src/main/scala/fi/oph/koski/validation/VSTKotoutumiskoulutus2022Validation.scala

```diff
@@ -0,0 +1,98 @@
+
+import fi.oph.koski.http.{HttpStatus, KoskiErrorCategory}
+import fi.oph.koski.schema._
+
+object VSTKotoutumiskoulutus2022Validation {
+  def validate(suoritus: Suoritus): HttpStatus =
+    suoritus match {
+      case s: OppivelvollisilleSuunnattuMaahanmuuttajienKotoutumiskoulutuksenSuoritus2022 =>
+        HttpStatus.fold(s.osasuoritukset.getOrElse(List.empty).map(validate))
+      case _ =>
+        HttpStatus.ok
+    }
+
+  def validate(osasuoritus: VSTKotoutumiskoulutuksenKokonaisuudenSuoritus2022): HttpStatus =
+    osasuoritus match {
+      case s: VSTKotoutumiskoulutuksenKieliJaViestintäosaamisenSuoritus2022 => validateKieliJaViestintä(s)
+      case s: VSTKotoutumiskoulutuksenYhteiskuntaJaTyöelämäosaaminenSuoritus2022 => validateYhteiskuntaJaTyöosaaminen(s)
+      case s: VSTKotoutumiskoulutuksenOhjauksenSuoritus2022 => valitedateOhjausLaajuus(s)
+      case _ => HttpStatus.ok
+    }
+
+  def validateKieliJaViestintä(osasuoritus: VSTKotoutumiskoulutuksenKieliJaViestintäosaamisenSuoritus2022): HttpStatus =
+    HttpStatus.fold(
+      validateKieliJaViestintäLaajuus(osasuoritus),
+      validateKieliJaViestintäOsasuoritustenArviointi(osasuoritus),
+    )
+
+  def validateKieliJaViestintäLaajuus(suoritus: VSTKotoutumiskoulutuksenKieliJaViestintäosaamisenSuoritus2022): HttpStatus =
+    josArvioitu(suoritus) {
+      validateLaajuus(suoritus, 40)
+    }
+
+  def validateKieliJaViestintäOsasuoritustenArviointi(suoritus: VSTKotoutumiskoulutuksenKieliJaViestintäosaamisenSuoritus2022): HttpStatus =
+    josArvioitu(suoritus) {
+      val pakollisetKieliopintojenAlaosasuoritukset = Set(
+        "kuullunymmartaminen",
+        "luetunymmartaminen",
+        "puhuminen",
+        "kirjoittaminen"
+      )
+
+      val arvioidutOsasuoritustenTyypit =
+        suoritus
+          .osasuoritukset
+          .map(_
+            .filter(_.arvioitu)
+            .map(_.tyyppi.koodiarvo))
+
+      arvioidutOsasuoritustenTyypit match {
+        case Some(tyypit) if pakollisetKieliopintojenAlaosasuoritukset.forall(tyypit.contains) => HttpStatus.ok
+        case _ => KoskiErrorCategory.badRequest.validation.tila.osasuoritusPuuttuu("Kielten ja viestinnän osasuoritusta ei voi hyväksyä ennen kuin kaikki pakolliset alaosasuoritukset on arvioitu")
+      }
+    }
+
+  def validateYhteiskuntaJaTyöosaaminen(suoritus: VSTKotoutumiskoulutuksenYhteiskuntaJaTyöelämäosaaminenSuoritus2022): HttpStatus =
+    HttpStatus.fold(
+      validateYhteiskuntaJaTyöosaaminenLaajuus(suoritus),
+      validateTyössäoppimisjaksot(suoritus)
+    )
+
+  def validateYhteiskuntaJaTyöosaaminenLaajuus(suoritus: VSTKotoutumiskoulutuksenYhteiskuntaJaTyöelämäosaaminenSuoritus2022): HttpStatus =
+    josArvioitu(suoritus) {
+      validateLaajuus(suoritus, 20)
+    }
+
+  def validateTyössäoppimisjaksot(suoritus: VSTKotoutumiskoulutuksenYhteiskuntaJaTyöelämäosaaminenSuoritus2022): HttpStatus =
+    josArvioitu(suoritus) {
+      val työssäoppimisjaksojenYhteislaajuus = suoritus
```

Comment:
Mietin, että eikö näitä laajuuksia lasketa jo yhteen automaattisesti aiemmin: voisiko käyttää valmiiksi yhteenlaskettua laajuutta, eikä laskea tässä validaatiossa summaa uudestaan?